aptos-move/block-executor/src/scheduler.rs

```diff
@@ -779,6 +829,9 @@ impl Scheduler {
     /// Return false when the execution is halted.
     fn suspend(&self, txn_idx: TxnIndex, dep_condvar: DependencyCondvar) -> bool {
         let mut status = self.txn_status[txn_idx as usize].0.write();
+        if self.done_marker.load(Ordering::Acquire) {
```

Comment:
I checked, they already synchronize on the status.
halt acquires the status and sets to ExecutionHalted, and hence if suspend happens after that it will return false.
if suspend happens before, halt will observe Suspended & wake it up.

So we shouldn't need this check.